### PR TITLE
ci: enable doc tests

### PR DESCRIPTION
**Pull request by s8sato**
## Context

Doc tests are disabled by default [cargo-llvm-cov](https://github.com/taiki-e/cargo-llvm-cov?tab=readme-ov-file#known-limitations)

### Solution

Enable it

### Checklist

- [x] I've read [`CONTRIBUTING.md`](../CONTRIBUTING.md).
- [ ] All review comments have been resolved.
- [x] All CI checks pass.

### CODE REVIEW

File: .github/workflows/iroha2-dev-pr.yml
Comment (0x009922) [inline .github/workflows/iroha2-dev-pr.yml:1]: Running doctests with only `--all-features` seems sufficient to me. Or is it not?

**Review by 0x009922 — DISMISSED** [review]

File: .github/workflows/iroha2-dev-pr.yml
Comment (s8sato) [reply .github/workflows/iroha2-dev-pr.yml:1]: Agreed, what doc tests should ensure would be that they are not false code, not that they works under any features set

**Review by 0x009922 — DISMISSED** [review]

**Comment by s8sato** [comment]
Converted to draft as @mversic has another solution to the failing doc test, and I'd wait for his PR and rebase it

File: crates/iroha_telemetry_derive/Cargo.toml [hunk 1]
```diff
@@ -30,4 +30,6 @@ manyhow = { workspace = true }
 iroha_macro_utils = { workspace = true }
 
 [dev-dependencies]
+iroha_core = { workspace = true }
+
```
Comment (mversic) [inline crates/iroha_telemetry_derive/Cargo.toml:34]: remove this

Comment (mversic) [inline crates/iroha_telemetry_derive/Cargo.toml:34]: remove this

**Review by mversic — CHANGES_REQUESTED** [review]

**Review by 0x009922 — APPROVED** [review]

**Pull request merged**